core_backend/tests/api/test_import_content.py

```diff
@@ -0,0 +1,265 @@
+def data_missing_columns() -> BytesIO:
+    data = {
+        "wrong_column_1": ["Value 1", "Value 2"],
+        "wrong_column_2": ["Value 3", "Value 4"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_title_missing() -> BytesIO:
+    data = {
+        "content_title": ["", "csv text 1"],
+        "content_text": ["csv title 2", "csv text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_text_missing() -> BytesIO:
+    data = {
+        "content_title": ["csv title 1", "csv title 2"],
+        "content_text": ["", "csv text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_long_title() -> BytesIO:
+    data = {
+        "content_title": ["a" * 151],
+        "content_text": ["Valid text"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_long_text() -> BytesIO:
+    data = {
+        "content_title": ["Valid title"],
+        "content_text": ["a" * 2001],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_duplicate_titles() -> BytesIO:
+    data = {
+        "content_title": ["Duplicate title", "Duplicate title"],
+        "content_text": ["Text 1", "Text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_duplicate_texts() -> BytesIO:
+    data = {
+        "content_title": ["Title 1", "Title 2"],
+        "content_text": ["Duplicate text", "Duplicate text"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_title_in_db() -> BytesIO:
+    # Assuming "Title in DB" is a title that exists in the database
+    data = {
+        "content_title": ["Title in DB"],
+        "content_text": ["New text"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_text_in_db() -> BytesIO:
+    # Assuming "Text in DB" is a text that exists in the database
+    data = {
+        "content_title": ["New title"],
+        "content_text": ["Text in DB"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+class TestImportContent:
+    async def test_csv_import_success(
+        self,
+        client: TestClient,
+        data_valid: BytesIO,
+        fullaccess_token: str,
+    ) -> None:
+        response = client.post(
+            "/content/csv-upload",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+            files={"file": ("test.csv", data_valid, "text/csv")},
+        )
+        assert response.status_code == 200
+
+        json_response = response.json()
+        for content in json_response:
+            content_id = content["content_id"]
+            response = client.delete(
+                f"/content/{content_id}",
+                headers={"Authorization": f"Bearer {fullaccess_token}"},
+            )
+            assert response.status_code == 200
+
+    @pytest.mark.parametrize(
+        "mock_csv_data, expected_error_type",
+        [
+            (
+                "data_empty_csv",
+                "empty_data",
```

Comment:
We should add these as a collection to avoid repetition.

Reply:
collection like collect them in a list beforehand? good idea